gems/pending/postgres_ha_admin/database_yml.rb

```diff
@@ -0,0 +1,68 @@
+require 'util/miq-password'
+
+module PostgresHaAdmin
+  class DatabaseYml
+    attr_reader :db_yml_file, :environment
+
+    def initialize(db_yml_file, environment)
+      @db_yml_file = db_yml_file
+      @environment = environment
+    end
+
+    def pg_params_from_database_yml
+      rails_params_to_pg(params_from_database_yml)
+    end
+
+    def update_database_yml(params)
+      db_yml = YAML.load_file(db_yml_file)
+      db_yml[environment] = params_from_database_yml
+      db_yml[environment].merge!(pg_parameters_to_rails(params))
+      remove_empty(db_yml[environment])
+
+      new_name = "#{db_yml_file}_#{Time.current.strftime("%d-%B-%Y_%H.%M.%S")}"
+      File.rename(db_yml_file, new_name)
+
+      File.write(db_yml_file, db_yml.to_yaml)
```

Comment:
If this fails we definitely wouldn't want to leave the machine with a bad database.yml

I would wrap this in some rescue logic to move the old file back, just in case something goes wrong.